### PR TITLE
feat: Allow resetting of the congestion controller state of a connection

### PR DESCRIPTION
**Pull request by flub**
This adds a method to reset the congestion controller state of a connection.  This is
useful if the underlying socket changed it's network path, like e.g. happens with the
iroh-net MagicSock.

### CODE REVIEW

File: iroh-quinn-proto/src/connection/paths.rs [hunk 1]
```diff
@@ -107,7 +107,7 @@ pub struct RttEstimator {
 }
 
 impl RttEstimator {
-    fn new(initial_rtt: Duration) -> Self {
+    pub(super) fn new(initial_rtt: Duration) -> Self {
```
Comment (dignifiedquire) [inline iroh-quinn-proto/src/connection/paths.rs:110]: this seems to not be needed anymore

File: iroh-quinn/src/connection.rs [hunk 2]
```diff
@@ -810,12 +815,24 @@ impl std::ops::Deref for ConnectionRef {
     }
 }
 
+/// Inner connection, use with care.
 #[derive(Debug)]
-pub(crate) struct ConnectionInner {
+pub struct ConnectionInner {
```
Comment (dignifiedquire) [inline iroh-quinn/src/connection.rs]: do you really need the whole thing, or could you just have sth like `WeakConnectionHandle(Weak<ConnectionInner>)` that implements`reset_congestion_state`?

File: iroh-quinn-proto/src/connection/paths.rs (hunk 1 above)
Comment (flub) [reply iroh-quinn-proto/src/connection/paths.rs:110]: This is still needed by the `reset_congestion_state` function

File: iroh-quinn/src/connection.rs (hunk 2 above)
Comment (flub) [reply iroh-quinn/src/connection.rs]: That's a good idea, I adopted this.

File: iroh-quinn-proto/src/connection/paths.rs (hunk 1 above)
Comment (dignifiedquire) [reply iroh-quinn-proto/src/connection/paths.rs:110]: right, I missed that

**Pull request merged**